api/v1alpha1/authorization_types.go

```diff
@@ -72,6 +76,20 @@ type Principal struct {
 	// JWT authentication in the same `SecurityPolicy`.
 	// +optional
 	JWT *JWTPrincipal `json:"jwt,omitempty"`
+
+	// Headers authorize the request based on the headers in the request.
```

Comment:
curious why we are choosing `AND` here
`OR` fits here better

or maybe we can't use `HeaderMatch` here and may need some other type

here is the use case, id like to deny traffic when `x-geoip-country` contains `foo` or `bar`
so maybe
```
headers:
- name: x-geo-ip
   values:
   - "foo"
   - "bar"  
```

fits better here
cc @nezdolik

Reply:
```
    authorization:
      defaultAction: Allow
      rules:
      - action: Deny
        principal:
          headers:
          - name: x-geoip-country
            value: foo
      - action: Deny
        principal:
          headers:
          - name: x-geoip-country
            value: bar
```
is too lengthy, the common case is there will be hundreds of such entries, which requires 5x more lines

Reply:
Ha, you're right! I was blind by trying to reuse the exiting HeaderMatch, instead, an array should be used for header value:

```
    authorization:
      defaultAction: Deny
      rules:
      - action: Allow
        principal:
          headers:
          - name: x-user-id
            values: [john, alice]
          - name: x-org-id
            values: sales
      - action: Deny
        principal:
          headers:
          - name: x-geoip-country
            values: [foo, bar]
```

This is similar to what we have done for JWT:

```
        principal:
          jwt:
            provider: example1
            scopes: [foo, bar]
```